src/sync.rs

```diff
@@ -623,6 +639,20 @@ impl<T: Copy> LockFreeFrozenVec<T> {
     }
 }
 
+impl<T: Copy + Clone> Clone for LockFreeFrozenVec<T> {
+    fn clone(&self) -> Self {
+        let cap = self.cap.load(Ordering::Acquire);
+        let len = self.len.load(Ordering::Acquire);
+
+        let new_vec = Self::with_capacity(cap);
+        for i in 0..len {
+            new_vec.push(self.get(i).unwrap());
```

Comment:
issue: to avoid overdoing atomic ops, we should first construct a `Vec<T>` and then wrap it at the end

Reply:
You mean something like this?
```rs

        let len = self.len.load(Ordering::Acquire);
        let cap = self.cap.load(Ordering::Acquire);
        // allocate the copy
        let layout = Self::layout(cap);
        let data = unsafe { std::alloc::alloc(layout).cast::<T>() };
        // copy the data
        unsafe {
            std::ptr::copy_nonoverlapping((*self.data.as_ptr()).cast::<T>(), data, len);
        }

        Self {
            data: AtomicPtr::new(data),
            len: AtomicUsize::new(len),
            cap: AtomicUsize::new(cap),
        }
    
```

Reply:
No need to call `alloc::alloc`, just use `Vec::into_raw_parts()`

Reply:
Re-implemented this for the new changes